### PR TITLE
Remove SVGs from JS bundle,

### LINKED ISSUE #498: Don't bundle SVG Images in JS build
## User story

Currently we have a number of SVG images that are bundled together in JS build output. This causes the JS build output significantly and unnecessarily large.

One way to fix it is to move those SVG images into a separate "public" / "assets" folder and don't include them in the JS build, and then we can display those images like below:

```
<img
	className="wc-admin-shipping-banner-illustration"
	src={ wcAdminAssetUrl + 'shippingillustration.svg' }
	alt={ __( 'Shipping ', 'woocommerce-admin' ) }
/>
```

Reference: https://github.com/woocommerce/woocommerce-admin/blob/5ebba741eaca7f73137db408996aa3c25d16612a/client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js#L400-L404

For more reference and implementations, checkout the WC-Admin repo and do a global search with the keyword ".svg".

Related slack thread: https://a8c.slack.com/archives/CK365S85V/p1618576095270200

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

- Remove SVGs from JS bundle, Use them as ``s.
- Add alt texts.
- Remove unnecessary width, height 100% styles.

Fixes https://github.com/woocommerce/google-listings-and-ads/issues/498

### Screenshots:

See below

### Detailed test instructions:

0. Before pulling this branch
1. Open the following cases in separate tabs to have something to compare `trunk` and `fix/498-svg-out`
2. Check it looks as before (in various window sizes)
3. Check the `alt` text is fine.

#### Cases:

1. **Marketing > GLA (Dashboard) > "Edit" Free Listings > step 2**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit&pageStep=2`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&subpath=%2Ffree-listings%2Fedit&pageStep=2)

	Hero image
	> Google Shopping search results example
2. **Marketing > GLA (Dashboard) > "Add paid campaign" > complete setup**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&guide=campaign-creation-success`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard&guide=campaign-creation-success)
	Successful guide
	> Drawing of a person who successfuly launched a campaign

3. Disconnect Google Account using Connection test page 
	[https://gla1.test/wp-admin/admin.php?page=connection-test-admin-page](https://gla1.test/wp-admin/admin.php?page=connection-test-admin-page)

4. **Marketing > GLA**
	[`/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&subpath=%2Freconnect-accounts&path=%2Fgoogle%2Fsettings)
	Re-connect Google
	> Google Logo
5. Disconnect all accounts at [`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsettings)
6. **Marketing > GLA**
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fstart)
Screenshot from 2021-07-21 18-13-02
	> Google Shopping search results example
	> Drawing of jigsaw puzzles connecting together
	> Drawing of a person looking at their mobile
	> Drawing of a bar and line charts heading up
7. Complete the setup
	[`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed&guide=submission-success`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fproduct-feed&guide=submission-success)
	Complete setup guide
	> WooCommerce Logo
	> Google Logo

### Changelog entry

> Tweak - Removed SVGs from JS bundle

### Additional notes:
1. I'm not sure if exposing `/js/src/...` as the assets directory is the safest and fits WordPress conventions. At least it does not require moving files around and lets us keep images close to components that use them.
2. I think that the ideal and the cheapest for an end-user solution would be to inline SVGs into HTML. However, AFAIK we do not use any SSR, and React component instances are 100% JS (as opposed to Custom Elements, which could be served as pure HTML for the first-page load).
3. I made up alt text by myself, so I'd appreciate an approval for the copy (in quote blocks below respective screenshots), @j111q?
4. [According to bundlewatch](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Cwoocommerce%7CrepoName%7Cgoogle-listings-and-ads%7CrepoCurrentBranch%7Cfix/498-svg-out%7CrepoBranchBase%7Ctrunk%7CcommitSha%7Cc0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7C./js/build/custom-inputs.js%7Csize%7CbaseBranchSize%7CmaxSize%7C./js/build/index.js%7C./js/build/product-attributes.js%7C./js/build/task-complete-setup.js%7C./js/build/index.css%7C./js/build/product-attributes.css%7C./google-listings-and-ads.zip%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B0B,%2B-745.75KB,%2B-7.0%2525)%5EKT%7CKT%7C2DC%7C7MT6%7CFSPT%7C1VF9C%7CD9%7CD9%7C2DC%7CWT%7CWT%7C2DC%7C53X%7C549%7C7WG%7C7Z%7C7Z%7CSG%7C5V9EI%7C63GQ1%7C7HP1C%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CS%7CI%7CT%7CJ%7CU%5D%7C$F%7CK%7CC%7CD%7CH%7CV%7CI%7CW%7CJ%7CX%5D%7C$F%7CL%7CC%7CD%7CH%7CY%7CI%7CZ%7CJ%7C10%5D%7C$F%7CM%7CC%7CD%7CH%7C11%7CI%7C12%7CJ%7C13%5D%7C$F%7CN%7CC%7CD%7CH%7C14%7CI%7C15%7CJ%7C16%5D%7C$F%7CO%7CC%7CD%7CH%7C17%7CI%7C18%7CJ%7C19%5D%7C$F%7CP%7CC%7CD%7CH%7C1A%7CI%7C1B%7CJ%7C1C%5D%5D%7CQ%7CR%5D%5D) our `index.js` bundle got down by ~52% (719,79KB -> 347.81KB)

### CODE REVIEW

**Review by eason9487 — COMMENTED** [review]
🎉  for the bundle size! 👍 

---

💅
 Looks like we don't need this SVG mock anymore:

https://github.com/woocommerce/google-listings-and-ads/blob/c0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b/jest.config.js#L23

(Just found that the [current test](https://github.com/woocommerce/google-listings-and-ads/blob/c0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b/tests/unit/wc-admin-dependency.test.js) did not cover the importing of SVG files, but there is no need to make up then.)

---

🚧 
The archived zip file would not include `js/src/` directory.

> 1. I'm not sure if exposing `/js/src/...` as the assets directory is the safest and fits WordPress conventions. At least it does not require moving files around and lets us keep images close to components that use them.

Probably we have to move all SVGs. If keep SVGs at their current positions, it would need to list all SVGs' paths in composer config.

We could consider a solution of centralizing the assets files by below steps:

1. Create a `assets` directory at project root
2. Add `"!/assets"` to the *exclude* list for including the assets directory into the archived zip file
    https://github.com/woocommerce/google-listings-and-ads/blob/c0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b/composer.json#L62-L68
3. Replace the `assetsURL` value with `$this->get_plugin_url( 'assets/' )`
    https://github.com/woocommerce/google-listings-and-ads/blob/c0ecd30c113c42a2e13735f9d4cf0e50f8dc6f4b/src/Admin/Admin.php#L111-L113
4. Move SVGs to assets directory and then they could be accessed via `/wp-content/plugins/google-listings-and-ads/assets/[filename].svg` path
5. Run `npm run archive` to see if they're included into zip file as well

**Comment by eason9487** [comment]
> 1. Create a `assets` directory at project root

Just noticed that the path `/wp-content/plugins/google-listings-and-ads/assets/google-logo.svg` will be blocked by some ad-block browser extensions. 🤦‍♂️   And `images` wouldn't be blocked.

File: js/src/get-started-page/features-card/index.js [hunk 1]
```diff
@@ -12,17 +12,31 @@ import {
 /**
  * Internal dependencies
  */
-import { ReactComponent as ImageConnection } from './img-connection.svg';
-import { ReactComponent as ImageFreeListings } from './img-free-listings.svg';
-import { ReactComponent as ImageGoogldAds } from './img-googld-ads.svg';
```
Comment (eason9487) [inline js/src/get-started-page/features-card/index.js:17]: Sorry for my typo 😂

File: js/src/components/account-card/index.js [hunk 2]
```diff
@@ -23,7 +32,14 @@ export const APPEARANCE = {
 
 const appearanceDict = {
 	[ APPEARANCE.GOOGLE ]: {
-		Logo: GoogleLogo,
+		Logo: (
```
Comment (eason9487) [inline js/src/components/account-card/index.js]: 💅 
How about rename to `logo` since it's changed to an instantiated component?

Comment (tomalec) [reply js/src/components/account-card/index.js]: This made me think, that serving an instance is not the best solution as if we are going to create two instances of `<AccountCard appearance="google">`, the Logo instance would not be duplicated and appear only in one. Do I get JSX right? I just got used to using `HTMLTemplateElement`s and instantiating them, and JSX is not a `<template>` ;)

Comment (tomalec) [reply js/src/components/account-card/index.js]: hm... :confused: it seems to eventually stamp two separate instances of ``.

Comment (tomalec) [reply js/src/components/account-card/index.js]: Addressed in https://github.com/woocommerce/google-listings-and-ads/pull/897/commits/1ead489dd787acd65fedb8d7541af71a23d04594

**Comment by tomalec** [comment]
> :construction:
> The archived zip file would not include js/src/ directory.

Thank you for catching this! :pray: 

I'd like to try to keep the files where they are, as it reduced the need for manual syncing assets used by components => reduced the costs and risk of maintaining abandoned files.

I decided [to try](https://github.com/woocommerce/google-listings-and-ads/pull/897/commits/c5de2c3a32944c882a250a2e8bd1d84178dfa485) this path.

> If keep SVGs at their current positions, it would need to list all SVGs' paths in composer config.

by adding `"!/js/src/**/*.svg",` to `composer.json`.

WDYT about it, does it pass ad blockers you checked?

**Comment by tomalec** [comment]
I think it's ready for another review.

File: js/src/components/account-card/index.js (hunk 2 above)
Comment (eason9487) [reply js/src/components/account-card/index.js]: Sorry, I used an imprecise term. It should be called **react element** if follow [the definitions of React official blog](https://reactjs.org/blog/2015/12/18/react-components-elements-and-instances.html#elements-describe-the-tree). And the **instance** should be referred to the component instance in runtime and it keeps references to relevant DOM.

**Comment by eason9487** [comment]
❓ 

> I'd like to try to keep the files where they are, as it reduced the need for manual syncing assets used by components => reduced the costs and risk of maintaining abandoned files.

#### Abandoned files maintainability

By declaring SVGs in constants or another dedicated file, and export them directly, the maintenance cost should be very small.

```js
// constants.js
const { assetsURL } = glaData;

export const svgGoogleLogo = assetsURL + 'gogole-logo.svg';
export const svgHero = assetsURL + 'google-free-listing.svg';

// Consumer side

import { svgGoogleLogo } from '.~/constants';
```

When checking whether a file is still in use, we could:
1. Global search the constant `\bsvgGoogleLogo\b` if editor supports regexp search
2. Comment out `export` to see if a warning shows up from the webpack building
    image

#### Code maintainability

Regardless of whether the variables are declared in the components that use them, or in the *constants.js*, only the filenames are required, or at most additional subpath in the assets directory.

Compare with the current method, it can eliminate the need for comments scattered in several places and the need to write the file paths under *js/src* individually as below.

```js
/**
 * Full URL to the Google G logo image.
 * Preferably we would inline it into HTML to reduce the bundle size.
 *
 * Unfortunately, React does not support `import.meta`, so we need to hardcode the module path.
 */
const googleLogoURL =
	glaData.assetsURL + 'js/src/components/account-card/gogole-g-logo.svg';
```

---

In terms of overall maintainability, such as adding new images (not only svg, maybe png or jpg, etc), moving files, and those additional comments, I think it's worth it to do a one-time move. WDYT?

**Comment by eason9487** [comment]
💅 
Thanks for adding this https://github.com/woocommerce/google-listings-and-ads/pull/897/commits/c101d69bae986b6181604606b9b0611867348bb8. I have one more small nitpick that the *tests/mocks/assets/svgrMock.js* file could be removed as well.

**Comment by tomalec** [comment]
> #### Abandoned files maintainability
>  […]
> When checking whether a file is still in use, we could:

Sorry, I didn't state that clearly, but my goal was to remove the need of "checking whether a file is still in use" totally. Consider the scenario: 

It happens that `<AccountCard>` is no longer needed. As a developer, all I need to do is:

1. Remove the `js/src/components/account-card/` folder

(our linters will make sure nothing is importing anything from there)

As opposed to the solution you propose where

1. Remove the `js/src/components/account-card/` folder.
2. Check for any images imported by **any file** in `js/src/components/account-card/`
	  1. For **every image**:
		  1. Global search the constant `\bvariableName\b` if editor supports regexp search
		  2. Comment out `export` to see if a warning shows up from the webpack building
		  3. Remove the `export` from '.~/constants' if there is nobody else using them

So to me "By declaring SVGs in constants or another dedicated file, and export them directly, the maintenance cost" when removing the component would increase from `O(1)` to `O(n^2)`


> #### Code maintainability
> 
> Regardless of whether the variables are declared in the components that use them, or in the *constants.js*, only the filenames are required, or at most additional subpath in the assets directory.

Sure, only filenames are required. But in my opinion, additional folder structure gives valuable information. By putting everything to a single flat bag of `/images` we would reduce discoverability and identification of images. Just like we don't put all `*.js` files in a single `/js` folder, we shouldn't put all `*.(svg|png)` files in a single folder. Especially, that none of those images is shared as itself. Only the components are shared.



> Compare with the current method, it can eliminate the need for comments scattered in several places and the need to write the file paths under *js/src* individually as below.

I agree. Moreover, this comment might have some sense for a single-use, but repeating it everywhere, and complaining that React does not follow ECMA standards, and produces JS instead of HTML, does not bring any value. I'll remove them.

**Comment by tomalec** [comment]
Another thing to note in favor of `~/js/src/components/**/*.(svg|png)` and against `/images/*.(svg|png)` is that:
- it keeps files closer to where they are used,
- makes component more atomic, independent, and self-sufficient

**Review by eason9487 — APPROVED** [review]
I have got @tomalec's perspective from https://github.com/woocommerce/google-listings-and-ads/pull/897#issuecomment-887365993. Considering that the current implementation has already addressed the issue #498, I think there's no blocker for now.

**Comment by eason9487** [comment]
@tomalec

🚧 

Sorry, I just noticed that the  **google-listings-and-ads.zip** file should be about the same size as the original, but [the reduced file size is almost the same](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7Cwoocommerce%7CrepoName%7Cgoogle-listings-and-ads%7CrepoCurrentBranch%7Cfix/498-svg-out%7CrepoBranchBase%7Ctrunk%7CcommitSha%7Cc5de2c3a32944c882a250a2e8bd1d84178dfa485%7Cresults%7Cstatus%7Cpass%7CfullResults%7CfilePath%7C./js/build/custom-inputs.js%7Csize%7CbaseBranchSize%7CmaxSize%7C./js/build/index.js%7C./js/build/product-attributes.js%7C./js/build/task-complete-setup.js%7C./js/build/index.css%7C./js/build/product-attributes.css%7C./google-listings-and-ads.zip%7Csummary%7CEverything%2Bis%2Bin%2Bcheck%2B(%252B0B,%2B-745.79KB,%2B-7.0%2525)%5EKT%7CKT%7C2DC%7C7MSX%7CFSPL%7C1VF9C%7CD9%7CD9%7C2DC%7CWT%7CWT%7C2DC%7C53X%7C549%7C7WG%7C7Z%7C7Z%7CSG%7C5VBA4%7C63IMK%7C7HP1C%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CS%7CI%7CT%7CJ%7CU%5D%7C$F%7CK%7CC%7CD%7CH%7CV%7CI%7CW%7CJ%7CX%5D%7C$F%7CL%7CC%7CD%7CH%7CY%7CI%7CZ%7CJ%7C10%5D%7C$F%7CM%7CC%7CD%7CH%7C11%7CI%7C12%7CJ%7C13%5D%7C$F%7CN%7CC%7CD%7CH%7C14%7CI%7C15%7CJ%7C16%5D%7C$F%7CO%7CC%7CD%7CH%7C17%7CI%7C18%7CJ%7C19%5D%7C$F%7CP%7CC%7CD%7CH%7C1A%7CI%7C1B%7CJ%7C1C%5D%5D%7CQ%7CR%5D%5D) as the **index.js**. After running `npm run archive` on my end, the archive file doesn't include the `src/js` directory. I suspected that the rule of `archive.exclude` doesn't support `**` to find files recursively.

https://github.com/woocommerce/google-listings-and-ads/blob/55ba6700b73baac168bf65b639468d21176337b9/composer.json#L70

After deeper investigation about `composer archive`, here are my findings:

- It applies rules in **.gitignore** to exclude files.
- It excludes files in **.gitattributes** which are marked with `export-ignore`.
- It applies the `archive.exclude` rules in  **composer.json** to exclude files, or to include files forcibly if the rule starts with `!`.
    - `composer` converts the `archive.exclude` glob rules to RegExp via `symfony/finder` library.
        https://github.com/composer/composer/blob/2.1.5/src/Composer/Package/Archiver/BaseExcludeFilter.php#L123-L147
    - The required `symfony/finder` version of `composer` spans several different major versions.
        https://github.com/composer/composer/blob/2.1.5/composer.json#L37
    - Newer versions of `symfony/finder` support `**` but the latest `composer.phar` is still using `symfony/finder` v2.8.52. And I think it is the root cause that `**` is not working.
      https://github.com/symfony/Finder/blob/v2.8.52/Glob.php

I haven't look into it but seems the [file-loader](https://webpack.js.org/loaders/file-loader/) might be another solution if want to keep images stay with needed components.

**Comment by tomalec** [comment]
@eason9487 I'm sorry, I missed it again :(
Thank you for finding that and an awesome investigation.

**Comment by tomalec** [comment]
To add to your investigation `composer archive` does not allow to set `$strictWildcardSlash = false`, to allow to mimic recursive wildcard https://github.com/symfony/Finder/blob/5.3/Glob.php#L79.

Thanks for finding `file-loader` -> [`asset-modules`](https://webpack.js.org/guides/asset-modules/).
Usually, I'm against over-webpackign everything. Especially, using more webpack to solve problems created by using webpack. (If we would write build-less, or at least webpack-less, and use vanilla JS ES Modules, we could use `import.meta` to get the current path).

But, given where we are, this seems to be a nice solution for our problem:
- lets us keep source files close to the code,
- could preserve or change the path of distributed assets
- could generate a content hash, which would allow merchants to enable strict caching rules.
	That hurts my old-school web-dev sentiment of a browser running same files I use during development. And could make production debugging a tad complicated.
	We can emit paths using the original path and filename, like: `/images/{path}/{contenthash}.{filename}.{ext}`. That way the files would be still matchable by a human, and the merchant would gain caching possibilities.
	
The only doubt I have is that we are still on Webpack 4, so we are forced to use something that we now are deprecated and has a cleaner alternative in the future. That kind of introduces additional technical debt.

**Pull request merged**